### PR TITLE
Upgrades Kafka version and Spring-kafka-test version to latest suggested by whitesource

### PR DESCRIPTION
**Pull request by DarshitChanpura**
Signed-off-by: Darshit Chanpura <dchanp@amazon.com>

### Description
Upgrades Kafka and spring-kafka-test jar versions
* Enhancement
* Failing Whitesource check

### Issues Resolved
https://github.com/opensearch-project/security/issues/1564

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -523,6 +524,22 @@
             <version>1.0-1</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
```
Comment (davidlago) [inline pom.xml:528]: What's this for? can we remove it?

Comment (DarshitChanpura) [reply pom.xml:528]: yes it is... these are the brokers that enable spring kafka to use 2.8.1...more detail can be found here: https://giters.com/spring-projects/spring-kafka/issues/1792

i tested it to make sure that it works correctly and is required

**Review by cliu123 — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -81,12 +81,13 @@
...
         <http.commons.version>4.5.13</http.commons.version>
         <cxf.version>3.4.5</cxf.version>
-        <kafka.version>2.5.0</kafka.version>
+        <kafka.version>2.8.1</kafka.version>
         <java-saml.version>2.5.0</java-saml.version>
         <opensaml.version>3.4.5</opensaml.version>
         <commons-codec.version>1.14</commons-codec.version>
         <xmlsec.version>2.2.3</xmlsec.version>
         <cryptacular.version>1.1.4</cryptacular.version>
+        <spring.kafka.version>2.7.9</spring.kafka.version>
```
Comment (cliu123) [inline pom.xml:90]: Why are we upgrading to`2.7.9` instead of the latest version `2.8.2`?

File: pom.xml [hunk 3]
```diff
@@ -81,12 +81,13 @@
         <ldaptive.version>1.2.3</ldaptive.version>
         <http.commons.version>4.5.13</http.commons.version>
         <cxf.version>3.4.5</cxf.version>
-        <kafka.version>2.5.0</kafka.version>
+        <kafka.version>2.8.1</kafka.version>
```
Comment (cliu123) [inline pom.xml:84]: Why are we not upgrading to the latest version `3.1.0`?

Comment (DarshitChanpura) [reply pom.xml:84]: need to test compatibility... 2.8.1 is what was suggested by whitesource...
and so spring-kafka is updated to 2.7.9

**Pull request merged**